core/trino-main/src/main/java/io/trino/execution/QueryTracker.java

```diff
@@ -311,6 +311,38 @@ int getPrunedQueriesCount()
         return prunedQueriesCount.get();
     }
 
+    @Managed
+    long getAllQueriesRetainedSize()
+    {
+        long result = 0;
+        for (T query : queries.values()) {
+            result += query.getRetainedSizeInBytes();
```

Comment:
is it expensive to call this?
will it consume a lot of CPU on the coordinator to serve this metric?

cc @lukasz-stec

Reply:
It is possible `QueryTracker` keeps thousands of queries and it looks like `getRetainedSizeInBytes` goes to `QueryStateMachine` and `QueryInfo` and recursively deeper. This has the potential to use some visible CPU  cycles. Would be good to benchmark it.